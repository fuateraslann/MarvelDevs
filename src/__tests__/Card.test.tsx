import { render } from '@testing-library/react'
import CardComponent from 'components/Card'

describe('CardComponent', () => {
  const mockProps = {
    imgSrc: 'test-image.jpg',
    name: 'Test Name',
    contentStyle: { backgroundColor: 'red' },
    extras: <div>extras</div>,
  }

  test('renders CardComponent correctly', () => {
    const { asFragment, getByAltText, getByText } = render(
      <CardComponent {...mockProps} />
    )
    const image = getByAltText('Test Name')
    expect(image).toBeInTheDocument()
    expect(getByText('Test Name')).toBeInTheDocument()
    const content = getByText('Test Name').closest('div')
    expect(content).toHaveStyle('background-color: red')
    expect(getByText('extras')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot() // Snpashot test
  })

  test('renders CardComponent with default content style if not provided', () => {
    const { getByText, asFragment } = render(
      <CardComponent imgSrc="test-image.jpg" name="Test Name" />
    )

    const content = getByText('Test Name').closest('div')
    expect(content).toHaveStyle(
      'height: 120px; padding: 5px; display: flex; justify-content: space-between;'
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders CardComponent without extras', () => {
    const { queryByTestId, asFragment } = render(
      <CardComponent imgSrc="test-image.jpg" name="Test Name" />
    )
    expect(queryByTestId('extras')).toBeNull()
    expect(asFragment()).toMatchSnapshot()
  })
})
