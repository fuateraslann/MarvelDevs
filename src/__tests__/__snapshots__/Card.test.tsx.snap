// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders CardComponent correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="height: 100%;"
  >
    <img
      alt="Test Name"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="90%"
      src="test-image.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      style="background-color: red;"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        Test Name
      </h6>
      <div>
        extras
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardComponent renders CardComponent with default content style if not provided 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="height: 100%;"
  >
    <img
      alt="Test Name"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="90%"
      src="test-image.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      style="height: 120px; padding: 5px; display: flex; justify-content: space-between;"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        Test Name
      </h6>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardComponent renders CardComponent without extras 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="height: 100%;"
  >
    <img
      alt="Test Name"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="90%"
      src="test-image.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      style="height: 120px; padding: 5px; display: flex; justify-content: space-between;"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        Test Name
      </h6>
    </div>
  </div>
</DocumentFragment>
`;
